apps/test/unit/lib/jsdebugger/JsDebuggerTest.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import sinon from 'sinon';
+import {Provider} from 'react-redux';
+import {mount} from 'enzyme';
+import {expect} from '../../../util/configuredChai';
+import JsDebugger, {UnconnectedJsDebugger} from '@cdo/apps/lib/jsdebugger/JsDebugger';
+import JsDebuggerUi from '@cdo/apps/lib/jsdebugger/JsDebuggerUi';
+import {getStore, registerReducers, stubRedux, restoreRedux} from '@cdo/apps/redux';
+import commonReducers from '@cdo/apps/redux/commonReducers';
+import {setPageConstants} from '@cdo/apps/redux/pageConstants';
+
+describe('The JSDebugger component', () => {
+  let root, jsDebugger;
+
+  beforeEach(() => {
+    stubRedux();
+    registerReducers(commonReducers);
+
+    const codeTextbox = document.createElement('div');
+    document.body.appendChild(codeTextbox);
+    codeTextbox.id = 'codeTextbox';
+
+    const runApp = sinon.spy();
+    getStore().dispatch(setPageConstants({
+      showDebugButtons: true,
+      showDebugConsole: true,
+      showDebugWatch: true,
+      showDebugSlider: true,
+      debuggerUi: new JsDebuggerUi(runApp, getStore()),
+    }));
+    root = mount(
+      <Provider store={getStore()}>
+        <JsDebugger style={{height: 250}}/>
+      </Provider>
+    );
+    jsDebugger = root.find('UnconnectedJsDebugger').get(0);
+  });
+
+  afterEach(() => {
+    restoreRedux();
+  });
+
+  const debugAreaEl = () => root.find('#debug-area').get(0);
+  const paneHeader = () => root.find('PaneHeader');
+  const closeIcon = () => paneHeader().find('i.fa-chevron-circle-down');
+  const openIcon = () => paneHeader().find('i.fa-chevron-circle-up');
```

Comment:
the const definitions are not in beforeEach because they never change. And they are functions because their return values might change depending on what actions have been taken. For example:

```jsx
closeIcon(); // returns a ReactWrapper with the close icon inside it.
closeIcon().simulate('click'); // changes state which makes the closeIcon go away
closeIcon(); // returns an empty ReactWrapper since the closeIcon is no longer there
```

if I instead has a `closeIcon` variable at the top level which wasn't a function, then i'd have to remember to reset it's value after calling `closeIcon.simulate('click')`, which would be more error prone.